torch/optim/optimizer.py

```diff
@@ -302,22 +338,28 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     @staticmethod
-    def _group_tensors_by_device_and_dtype(tensorlistlist, with_indices=False):
+    def _group_tensors_by_device_and_dtype(
+        tensorlistlist: List[List[Optional[torch.Tensor]]],
+        with_indices: bool = False,
+    ) -> Union[
+        Dict[Tuple[None, None], Tuple[List[List[Optional[torch.Tensor]]], List[int]]],
```

Comment:
LOL could we do some TypeAliasing here too, e.g.,

TensorListList would be List[List[Optional[torch.Tensor]]]
Indices would be List[int]